sdk/go/common/resource/plugin/context.go

```diff
@@ -103,6 +104,15 @@ func NewContextWithContext(
 		statusD = diag.DefaultSink(io.Discard, io.Discard, diag.FormatOptions{Color: colors.Never})
 	}
 
+	var projectName tokens.PackageName
+	projPath, err := workspace.DetectProjectPath()
```

Comment:
I don't love this, but since pulumi-yaml uses NewContextWithRoot, we can't add a new argument to it. We also do something similar in NewContext above for the same reason.

This works for project name, but if we'd like to pass the stack and organization names through to the dynamic provider, we have to find a different way of doing this.